src/sietch/distributed_dune.ml

```diff
@@ -152,51 +173,48 @@ let prefetch ({ cache; _ } as t) count i key =
     else
       let hash = Digest.to_string key in
       let () = debug [ Pp.textf "fetch metadata %s (%i/%i)" hash i count ] in
-      block_get t key >>= function
-      | None -> Async.Deferred.Result.return ()
-      | Some (metadata_contents, _) -> (
-        let* metadata =
-          Cache.Local.Metadata_file.of_string metadata_contents |> Async.return
+      let* _, metadata_contents = block_get t key in
+      let* metadata_contents =
+        Async.Pipe.fold ~init:""
+          ~f:(fun l r -> Async.return (l ^ r))
```

Comment:
Are we streaming metadata too? It doesn't seem necessary as metadata files are small, but I guess this may be a trade-off to reduce complexity, in which case it deserves a comment.

Reply:
Aha, you already answered this question. Still, it's worth adding a note about why we're doing this.